### PR TITLE
feat: add BLOCKSCOUT_HOST, and use it in API docs

### PR DESCRIPTION
**Pull request by zachdaniel**
## Motivation

* Right now the host is wrong in the API documentation page. 

### Enhancements
* This uses the correct value to render the host in api docs, and allows it to be configured via `BLOCKSCOUT_HOST` env variable.

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so if necessary

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/templates/api_docs/index.html.eex [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div class="card">
     <div class="card-body">
       <h1 class="card-title margin-bottom-sm"><%= gettext("API Documentation") %></h2>
-      <p class="api-text-monospace" data-endpoint-url="<%= BlockScoutWeb.Endpoint.url() %>/api">[ <%= gettext "Base URL:" %> <%= @conn.host %>/api ]</p>
+      <p class="api-text-monospace" data-endpoint-url="<%= BlockScoutWeb.Endpoint.url() %>/api">[ <%= gettext "Base URL:" %> <%= BlockScoutWeb.Endpoint.url() %>/api ]</p>
```
Comment (ayrat555) [inline apps/block_scout_web/lib/block_scout_web/templates/api_docs/index.html.eex]: Will it include a port in the URL?

**Comment by vbaranov** [comment]
@zachdaniel 

Port is not returned in the api docs page for Eth RPC endpoints
Screenshot 2019-06-20 at 10 56 47

File: apps/block_scout_web/lib/block_scout_web/templates/api_docs/index.html.eex (hunk 1 above)
Comment (zachdaniel) [reply apps/block_scout_web/lib/block_scout_web/templates/api_docs/index.html.eex]: Interesting, I thought that I had confirmed that port was present in that case. I'll take a look.

Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/templates/api_docs/index.html.eex]: I see port in the api docs page for JSON RPC, but it is missing in ETH JSON RPC page

Comment (zachdaniel) [reply apps/block_scout_web/lib/block_scout_web/templates/api_docs/index.html.eex]: Okay, port is included in that already, yes.

**Comment by zachdaniel** [comment]
@vbaranov if you merge the other PR for the compatibility matrix, I can include this change for that page.

**Comment by vbaranov** [comment]
> @vbaranov if you merge the other PR for the compatibility matrix, I can include this change for that page in this PR.

@zachdaniel #2167 has been merged

**Pull request merged**